x/datadeal/types/message_deal.go

```diff
@@ -16,27 +16,20 @@ func (m *MsgCreateDeal) Type() string {
 }
 
 func (m *MsgCreateDeal) ValidateBasic() error {
-	_, err := sdk.AccAddressFromBech32(m.ConsumerAddress)
-	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid consumer address (%s)", err)
-	}
-
-	schema := m.DataSchema
-	if len(schema) == 0 {
+	if len(m.DataSchema) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "no data schema")
 	}
-
-	budget := m.Budget
-	if budget == nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "budget is empty")
+	if m.MaxNumData <= 0 {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "max num of data is negative number")
+	}
+	if m.Budget == nil {
```

Comment:
It seems that a negative check of the budget should also be done.

Reply:
It was already included in `m.Budget.IsValid()` :)